source/agora/consensus/Quorum.d

```diff
@@ -0,0 +1,1118 @@
+                scope (exit) writefln("Intersection check for %s %s took %s",
+                    line, rand_idx, sw.peek.seconds);
+            }
+
+            verifyQuorumsIntersect(quorums);
+        }
+
+        foreach (key, quorum; quorums)
+        {
+            foreach (node; quorum.nodes)
+                counts[node]++;
+        }
+    }
+
+    return counts;
+}
+
+/// Generate random seeds by hashing a range of numbers: [0 .. count)
+private auto getRandSeeds (size_t count)
+{
+    // using 'ulong' to get consistent hashes
+    return iota(0, count).map!(idx => hashFull(cast(ulong)idx));
+}
+
+/// ditto
+private string pretty (PublicKey input)
+{
+    // convert arbitrary hashes into user-readable strings like "Andrew, Dave, etc"
+    static string toUserReadable (Hash hash)
+    {
+        const names =
+        [
+            "Aaron",
+            "Adam",
+            "Alex",
+            "Andrew",
+            "Anthony",
+            "Austin",
+            "Ben",
+            "Brandon",
+            "Brian",
+            "Charles",
+            "Chris",
+            "Daniel",
+            "David",
+            "Edward",
+            "Eric",
+            "Ethan",
+            "Fred",
+            "George",
+            "Iain",
+            "Jack",
+            "Jacob",
+            "James",
+            "Jason",
+            "Jeremy",
+            "John",
+            "Jonathan",
+            "Joseph",
+            "Josh",
+            "Justin",
+            "Kevin",
+            "Kyle",
+            "Luke",
+            "Mark",
+            "Martin",
+            "Mathew",
+            "Matthew",
+            "Michael",
+            "Nathan",
+            "Nicholas",
+            "Nick",
+            "Patrick",
+            "Paul",
+            "Peter",
+            "Philip",
+            "Richard",
+            "Robert",
+            "Ryan",
+            "Samuel",
+            "Scott",
+            "Sean",
+            "Simon",
+            "Stephen",
+            "Steven",
+            "Thomas",
+            "Timothy",
+            "Tyler",
+            "William",
+            "Zach",
+        ];
+
+        static size_t last_used;
+        static string[Hash] hashToName;
+
+        if (auto name = hash in hashToName)
+        {
+            return *name;
+        }
+        else
+        {
+            string name = names[last_used];
+            last_used++;
+
+            if (last_used >= names.length)
+                assert(0);  // need more names
+
+            hashToName[hash] = name;
+            return name;
+        }
+    }
+
+    return toUserReadable(input[].hashFull());
+}
+
+/// convenience because Amount's API sucks
+private Amount added (in Amount lhs, in Amount rhs)
+{
+    Amount amount = lhs;
+    assert(amount.add(rhs));
+    return amount;
+}
+
+/// convenience
+private Amount subbed (in Amount lhs, in Amount rhs)
+{
+    Amount amount = lhs;
+    assert(amount.sub(rhs));
+    return amount;
+}
```

Comment:
https://github.com/bpfkorea/agora/blob/20decfd4fed2b43a55e444c284d5a88d44ef0a75/source/agora/common/Amount.d#L184-L198

Reply:
Those modify the object. So I can't use them with const values.

E.g. I can't do `Amount.MinFreezeAmount.mustAdd(Amount.MinFreezeAmount)`.

Reply:
`Amount(Amount.MinFreezeAmount).mustAdd(Amount.MinFreezeAmount)` ?